test/scenarios.py

```diff
@@ -3810,4 +3812,222 @@
             ),
         },
     ),
+    Scenario(
+        "scenario23-1",
+        "Send two files with the same name but different fs location, expect them to transfer sucesfully",
+        {
+            "ren": ActionList(
+                [
+                    action.WaitForAnotherPeer(),
+                    action.NewTransfer(
+                        "172.20.0.15",
+                        ["/tmp/testfile-small", "/tmp/duplicate/testfile-small"],
+                    ),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                                event.File(
+                                    FILES["duplicate/testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.Wait(event.Start(0, FILES["testfile-small"].id)),
+                    action.Wait(
+                        event.FinishFileUploaded(
+                            0,
+                            FILES["testfile-small"].id,
+                        )
+                    ),
+                    action.Wait(event.Start(0, FILES["duplicate/testfile-small"].id)),
+                    action.Wait(
+                        event.FinishFileUploaded(
+                            0,
+                            FILES["duplicate/testfile-small"].id,
+                        )
+                    ),
+                    action.ExpectCancel([0], True),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+            "stimpy": ActionList(
+                [
+                    action.Wait(
+                        event.Receive(
+                            0,
+                            "172.20.0.5",
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
```

Comment:
I don't think `event.File` needs both ID fields anymore as only the first ID is enough

Reply:
Hmm,, you're right. Maybe putting those filenames in the `FILES` would make it more clear?

Reply:
Um still not true. For example, if you have a file `/tmp/foo/bar` then you have two options: either someone sends you `/tmp/foo` or `/tmp/foo/bar`. And you need to know that in the first case, you save the file as `foo/bar` and in the second case as `bar` in the destination directory.
Both of these cases are exercised in the integration tests, so you cannot assume in advance the file subpath used on a particular file in `config.py`